bpo-37001: Makes symtable.symtable have parity with compile for input

DinoV opened the pull request:
This PR modifies symtable.symtable() to allow bytes objects as source input similar to how they're accepted in compile().

Currently if you were parsing code with ast.parse(), and attempting to understand the name bindings with symtable.symtable(), you have one of two options:
    1) Only accept strings
    2) Use the tokenize module to go through a separate code path to attempt to parse the PEP 263 coding string.

Given that symtable.symtable ultimately goes down to the same underlying APIs as compile() it seems silly to require these extra steps.  So this unifies exec/eval/compile/symtable.symtable to use a single code path (_Py_SourceAsString) for doing the conversion.  This is the same as the existing source_as_string which is simply renamed and moved in with the other various parsing functions.

<!-- issue-number: [bpo-37001](https://bugs.python.org/issue37001) -->
https://bugs.python.org/issue37001
<!-- /issue-number -->

DinoV committed:
Makes symtable.symtable have parity for accepted datatypes
for source code as compile()

DinoV committed:
Add NEWS blurb

serhiy-storchaka commented on Modules/symtablemodule.c:
```diff
@@ -14,7 +14,7 @@ module _symtable
 _symtable.symtable
 
-    str:       str
+    str:       object
```
I suggest to rename `str` to `source`.

serhiy-storchaka commented on Modules/symtablemodule.c:
```diff
@@ -41,12 +51,15 @@ _symtable_symtable_impl(PyObject *module, const char *str,
+    }
     t = (PyObject *)st->st_top;
     Py_INCREF(t);
     PyMem_Free((void *)st->st_future);
```
Leaked `source_copy`.

DinoV committed:
Ensure source_copy is always decref'd
rename str->source - lso matching compile, after verifying
  we don't currently accept kw arguments

serhiy-storchaka commented on Misc/NEWS.d/next/Library/2019-05-23-21-10-57.bpo-37001.DoLvTK.rst:
```diff
@@ -0,0 +1,2 @@
+symtable.symtable now accepts the same input types for source code as the
```
```suggestion
:func:`symtable.symtable` now accepts the same input types for source code as the
```

serhiy-storchaka commented on Misc/NEWS.d/next/Library/2019-05-23-21-10-57.bpo-37001.DoLvTK.rst:
```diff
@@ -0,0 +1,2 @@
+symtable.symtable now accepts the same input types for source code as the
+built-in compile function.
```
```suggestion
built-in :func:`compile` function.
```

serhiy-storchaka committed:
Update 2019-05-23-21-10-57.bpo-37001.DoLvTK.rst

serhiy-storchaka reviewed (approved):
LGTM, but you need to regenerate generated files. `make regen-all`

DinoV committed:
Fix clinic generated code

DinoV committed:
Merge branch 'dinov/symtable' of https://github.com/DinoV/cpython into dinov/symtable
